### ISSUE TITLE
Legenda

### ISSUE THREAD

**Issue by mkyral** [osmcz/osmcz#49]
Chybí legenda. Ideálně navázaná na jednotlivé mapové podklady.

**Comment by severak** [comment]
Hezkou legendu poskytuje plugin [Leaflet.TileLegend](https://github.com/yohanboniface/Leaflet.TileLegend).

**Comment by mkyral** [comment]
Jo. To je pěkné. Stačí jen připravit ten [JSON soubor](https://github.com/yohanboniface/Leaflet.TileLegend#define-a-json-describing-your-legend)

**Comment by walley** [comment]
najdou lidi ten cudlik dole vubec?

**Comment by mkyral** [comment]
To nevím. Zřejmě bude potřeba to mírně upravit a změnit to na odkaz.

**Comment by zbycz** [comment]
Kartografové by vás hnali, legenda má vypadat spíš tak jak je na osm.org :-) Opět doporučuju využít [L.OSM.key](https://github.com/openstreetmap/openstreetmap-website/blob/master/app/assets/javascripts/leaflet.key.js)

![image](https://cloud.githubusercontent.com/assets/385047/12329883/b3fb84ec-bae2-11e5-9948-d1924ec13b4d.png)

**Comment by mkyral** [comment]
Tak tam přidáme otázku: Jste kartograf Ano / Ne? A podle odpovědi naservírujeme legendu :-D

V každém případě by se mi zamlouvalo, kdybych mohl kliknout a přeneslo by mně to na nějaký reálný příklad.

Mimochodem - "Main road" - kde je Dalibor? :-D

**Comment by zbycz** [comment]
Já jsem vlastně legendu na webové mapě ještě nikdy mepouzil/nepotřeboval.
Tak můj názor neberte tady moc v potaz.

Zkusil bys mi prosím popsat takový use-case, kdy se člověku hodí legenda?

**Comment by severak** [comment]
V tom případě spíš obdoba funkce _Co je zde?_, kterou mají mapy.cz. Když nebudu vědět, co daný symbol/vzor znamená, tak kliknu do mapy.

**Comment by zbycz** [comment]
K tomu usecasu - tj. jde mi o to: "nevím co to je a chci to zjistit"? (hezky to má map1.eu). Nebo mě zajímá co všechno v mapě je? Proč mě to zajímá - z jaké pozice? Dík!

**Comment by mkyral** [comment]
A nesežerou nás kartografové, když to tam nebude? :-D

My ty ikonky a čáry většinou známe, ale někdo úplně cizí si nemusí být úplně jistý, co na té mapě vlastně je. Například polní/lesní cesta vs. pěšina. Různé typy (grade1 - grade5) - sám si občas nejsem jistý co přesně to je.

Tohle se částečně vyřeší tou aktivní vrstvou, ale ta bude zatím jen na POI. Na čáry by jsi pak musel použít ten nástroj: Co je tady. 

(S tím problémem, že musíš klikat někde, k okraji, kde se čára lomí, protože když klikneš do plochy, nebo mezi dva body daleko od sebe, tak ti to nic nevrátí - vlastnost osm databáze)

**Comment by zbycz** [comment]
A jo, vidíš. To je moc dobrý postřeh. V tom případě by možná dávalo smysl zobrazit zejména prvky, které nejsou úplně jasné – jako ty právě zmiňované pěšími a polní cesty. Co myslíš?

**Comment by severak** [comment]
@zbycz Právě že ta verze od map1.eu je dost otravná. Až po kliknutí, vysvětlovat povrch na mouseover je overkill.

**Comment by zbycz** [comment]
@severak To souhlas - myslel jsem spíš graficky/informačně pěkná. Rozhodně souhlas že po kliknutí - když už.

**Comment by vop-cz** [comment]
...a/nebo zobrazovat podrobnosti prave jen pri otevrene legende, kdy se 
zajem uzivatele zjistovat podrobnosti predpoklada...

---------- Původní zpráva ----------

Od: Pavel Zbytovský 

Datum: 14. 1. 2016 v 23:26:54

Předmět: Re: [osmcz] Legenda (#49)

@severak(https://github.com/severak) To souhlas - myslel jsem spíš graficky/
informačně pěkná. Rozhodně souhlas že po kliknutí - když už.

—

Reply to this email directly or view it on GitHub
(https://github.com/osmcz/osmcz/issues/49#issuecomment-171801757). 

#

**Comment by jzvc** [comment]
2zbycz: Usecase ... zrovna cumim na sjezdo-bezkarskej overlay, a zajimalo by me, proc sou nad nekterejma trasama vykricniky v trojuhelniku ... a jeste navic nekde bily a nekde zluty ;D.

**Comment by zbycz** [comment]
Jj, takže zejména věci, které jsou nesrozumitelné? Aka má smysl tam
zobrazovat, že dálnice je dálnice, když ti je jasně poznat? Nevytváří to
jen informační šum?

ne 17. 1. 2016 v 12:17 odesílatel jzvc notifications@github.com napsal:

**Comment by jzvc** [comment]
To co je zjevny na prvni pohled bych do toho nedaval. Je treba to brat tak, ze tohle neni papirova mapa, a kdyby se ty data mely dostat na papir, tak to bud bude naprosto nepouzitelne cosi, nebo to budou desitky map se spoustou stranek popisek co kde co znamena.

Osobne bych to pojmul spis tak, ze legenda tematicky k prislusne vrstve (cimz myslim treba prave zmineny lyzarsky overlay). Ta by mohla byt pomerne strucna a uchopitelna.

**Comment by mkyral** [comment]
No koukám na stránky opensnowmap a legendu mají zpracovanou pěkně, jen by to chtělo ještě nějaký krátký vysvětlující textík.

![opm_legend](https://cloud.githubusercontent.com/assets/155542/12390041/27a03742-bddd-11e5-8469-2885b148c611.png)

Mimochodem. Ten jejich panel se mi docela líbí ;-) #28

**Comment by zbycz** [comment]
Když se člověk snaží, tak to jde pochopit i bez textu :D fakt dobré.
Jinak viděl bych to spíše vyjížděcí v sidebaru aka L.OSM.Sidebar .. v průběhu týden ho naimplementuju a pak se tam půjde dát vymýšlet leccos

**Comment by mkyral** [comment]
No alespoň title by ty obrázky mít mohly. Aby si člověk ověřil, že to fakt pochopil správně.

**Comment by zbycz** [comment]
Souhlas - zkopírujeme a rozhodně přidáme i minipopis.
